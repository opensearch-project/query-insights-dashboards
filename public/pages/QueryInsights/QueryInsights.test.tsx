/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import QueryInsights from './QueryInsights';
import { MemoryRouter } from 'react-router-dom';
import { MockQueries } from '../../../test/testUtils';

// Mock functions and data
const mockOnTimeChange = jest.fn();
const mockCore = {
  chrome: {
    setBreadcrumbs: jest.fn(),
  },
};

const sampleQueries = MockQueries();

const renderQueryInsights = () =>
  render(
    <MemoryRouter>
      <QueryInsights
        queries={sampleQueries}
        loading={false}
        onTimeChange={mockOnTimeChange}
        recentlyUsedRanges={[]}
        currStart="now-15m"
        currEnd="now"
        // @ts-ignore
        core={mockCore}
      />
    </MemoryRouter>
  );

describe('QueryInsights Component', () => {
  beforeAll(() => {
    jest.spyOn(Date.prototype, 'toLocaleTimeString').mockImplementation(() => '12:00:00 AM');
    jest.spyOn(Date.prototype, 'toDateString').mockImplementation(() => 'Mon Jan 13 2025');
  });

  afterAll(() => {
    jest.resetAllMocks(); // Reset all mocks after all tests
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the table with the correct columns and data', () => {
    const { container } = renderQueryInsights();
    expect(container).toMatchSnapshot();
  });

  it('calls setBreadcrumbs on mount', () => {
    renderQueryInsights();
    expect(mockCore.chrome.setBreadcrumbs).toHaveBeenCalledWith([
      {
        text: 'Query insights',
        href: '/queryInsights',
        onClick: expect.any(Function),
      },
    ]);
  });

  it('triggers onTimeChange when the date picker changes', () => {
    renderQueryInsights();

    // Find the date picker update button
    const updateButton = screen.getByRole('button', { name: /Refresh/i });
    fireEvent.click(updateButton);

    // Verify the onTimeChange callback is triggered
    expect(mockOnTimeChange).toHaveBeenCalled();
  });
});
