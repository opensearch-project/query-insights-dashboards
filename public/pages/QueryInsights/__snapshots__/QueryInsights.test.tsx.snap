// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryInsights Component renders the table with the correct columns and data 1`] = `
<div>
  <div>
    <div
      class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
    >
      <div
        class="euiFlexItem euiSearchBar__searchHolder"
      >
        <div
          class="euiFormControlLayout euiFormControlLayout--fullWidth"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <input
              aria-label="This is a search bar. After typing your query, hit enter to filter the results lower in the page."
              class="euiFieldSearch euiFieldSearch--fullWidth"
              placeholder="Search queries"
              type="search"
              value=""
            />
            <div
              class="euiFormControlLayoutIcons"
            >
              <span
                class="euiFormControlLayoutCustomIcon"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero euiSearchBar__filtersHolder"
      >
        <div
          class="euiFilterGroup"
        >
          <div
            class="euiPopover euiPopover--anchorDownCenter"
            id="field_value_selection_0"
          >
            <div
              class="euiPopover__anchor"
            >
              <button
                class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                    />
                  </svg>
                  <span
                    class="euiButtonEmpty__text"
                  >
                    <span
                      class="euiFilterButton__textShift"
                      data-text="Indices"
                      title="Indices"
                    >
                      Indices
                    </span>
                  </span>
                </span>
              </button>
            </div>
          </div>
          <div
            class="euiPopover euiPopover--anchorDownCenter"
            id="field_value_selection_1"
          >
            <div
              class="euiPopover__anchor"
            >
              <button
                class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                    />
                  </svg>
                  <span
                    class="euiButtonEmpty__text"
                  >
                    <span
                      class="euiFilterButton__textShift"
                      data-text="Search Type"
                      title="Search Type"
                    >
                      Search Type
                    </span>
                  </span>
                </span>
              </button>
            </div>
          </div>
          <div
            class="euiPopover euiPopover--anchorDownCenter"
            id="field_value_selection_2"
          >
            <div
              class="euiPopover__anchor"
            >
              <button
                class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                    />
                  </svg>
                  <span
                    class="euiButtonEmpty__text"
                  >
                    <span
                      class="euiFilterButton__textShift"
                      data-text="Coordinator Node ID"
                      title="Coordinator Node ID"
                    >
                      Coordinator Node ID
                    </span>
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiSuperDatePicker__flexWrapper"
        >
          <div
            class="euiFlexItem"
          >
            <div
              class="euiFormControlLayout euiFormControlLayout--group euiSuperDatePicker"
            >
              <div
                class="euiPopover euiPopover--anchorDownLeft"
              >
                <div
                  class="euiPopover__anchor euiQuickSelectPopover__anchor"
                >
                  <button
                    aria-label="Date quick select"
                    class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiFormControlLayout__prepend"
                    data-test-subj="superDatePickerToggleQuickMenuButton"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                    >
                      <svg
                        aria-hidden="true"
                        class="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                        focusable="false"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                        />
                      </svg>
                      <span
                        class="euiButtonEmpty__text euiQuickSelectPopover__buttonText"
                      >
                        <svg
                          aria-hidden="true"
                          class="euiIcon euiIcon--medium euiIcon-isLoading"
                          focusable="false"
                          height="16"
                          role="img"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                          />
                        </svg>
                      </span>
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiDatePickerRange euiDatePickerRange--inGroup"
                >
                  <button
                    class="euiSuperDatePicker__prettyFormat"
                    data-test-subj="superDatePickerShowDatesButton"
                  >
                    Last 15 minutes
                    <span
                      class="euiSuperDatePicker__prettyFormatLink"
                    >
                      Show dates
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <span
              class="euiToolTipAnchor"
            >
              <button
                class="euiButton euiButton--primary euiSuperUpdateButton"
                data-test-subj="superDatePickerApplyTimeButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                    />
                  </svg>
                  <span
                    class="euiButton__text euiSuperUpdateButton__text"
                  >
                    Refresh
                  </span>
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      class="euiBasicTable"
    >
      <div>
        <div
          class="euiTableHeaderMobile"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            />
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <div
                class="euiTableSortMobile"
              >
                <div
                  class="euiPopover euiPopover--anchorDownRight"
                >
                  <div
                    class="euiPopover__anchor"
                  >
                    <button
                      class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                      type="button"
                    >
                      <span
                        class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                      >
                        <svg
                          aria-hidden="true"
                          class="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                          focusable="false"
                          height="16"
                          role="img"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                          />
                        </svg>
                        <span
                          class="euiButtonEmpty__text"
                        >
                          Sorting
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <table
          class="euiTable euiTable--responsive"
          id="random_html_id"
          tabindex="-1"
        >
          <caption
            class="euiScreenReaderOnly euiTableCaption"
          />
          <thead>
            <tr>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_Timestamp_0"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Timestamp"
                    >
                      Timestamp
                    </span>
                  </span>
                </button>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_measurements_1"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Latency"
                    >
                      Latency
                    </span>
                  </span>
                </button>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_measurements_2"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="CPU Time"
                    >
                      CPU Time
                    </span>
                  </span>
                </button>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_measurements_3"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Memory Usage"
                    >
                      Memory Usage
                    </span>
                  </span>
                </button>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_indices_4"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Indices"
                    >
                      Indices
                    </span>
                  </span>
                </button>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_search_type_5"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Search Type"
                    >
                      Search Type
                    </span>
                  </span>
                </button>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_node_id_6"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Coordinator Node ID"
                    >
                      Coordinator Node ID
                    </span>
                  </span>
                </button>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_total_shards_7"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Total Shards"
                    >
                      Total Shards
                    </span>
                  </span>
                </button>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="euiTableRow"
            >
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Timestamp
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText euiTableCellContent--overflowingContent"
                >
                  <span
                    class=""
                  >
                    <button
                      class="euiLink euiLink--primary"
                      type="button"
                    >
                      Jan 13, 2025 @ 12:00:00 AM
                    </button>
                  </span>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Latency
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText euiTableCellContent--overflowingContent"
                >
                  8 ms
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  CPU Time
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText euiTableCellContent--overflowingContent"
                >
                  5.51 ms
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Memory Usage
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText euiTableCellContent--overflowingContent"
                >
                  Not enabled
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Indices
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText euiTableCellContent--overflowingContent"
                >
                  top_queries-2024.09.12
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Search Type
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText euiTableCellContent--overflowingContent"
                >
                  query then fetch
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Coordinator Node ID
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText"
                >
                  <span
                    class="euiTableCellContent__text"
                  >
                    Q36D2z_NRGKim6EZZMgi6A
                  </span>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Total Shards
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText"
                >
                  <span
                    class="euiTableCellContent__text"
                  >
                    1
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div>
        <div
          class="euiSpacer euiSpacer--m"
        />
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiPopover euiPopover--anchorUpRight"
            >
              <div
                class="euiPopover__anchor"
              >
                <button
                  class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall"
                  data-test-subj="tablePaginationPopoverButton"
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                  >
                    <svg
                      aria-hidden="true"
                      class="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                      focusable="false"
                      height="16"
                      role="img"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                      />
                    </svg>
                    <span
                      class="euiButtonEmpty__text"
                    >
                      Rows per page
                      : 
                      10
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <nav
              class="euiPagination"
            >
              <button
                aria-label="Previous page"
                class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall"
                data-test-subj="pagination-button-previous"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                  />
                </svg>
              </button>
              <ul
                class="euiPagination__list"
              >
                <li
                  class="euiPagination__item"
                >
                  <button
                    aria-controls="random_html_id"
                    aria-current="true"
                    aria-label="Page 1 of 1"
                    class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiButtonEmpty-isDisabled euiPaginationButton euiPaginationButton-isActive euiPaginationButton--hideOnMobile"
                    data-test-subj="pagination-button-0"
                    disabled=""
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButtonEmpty__content"
                    >
                      <span
                        class="euiButtonEmpty__text"
                      >
                        1
                      </span>
                    </span>
                  </button>
                </li>
              </ul>
              <button
                aria-label="Next page"
                class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall"
                data-test-subj="pagination-button-next"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                  />
                </svg>
              </button>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
